src/libNode/Node.cpp

```diff
@@ -435,14 +435,16 @@ bool Node::CheckIntegrity(bool continueOnError) {
       }
       if (BlockStorage::GetBlockStorage().GetMicroBlock(mbInfo.m_microBlockHash,
                                                         mbptr)) {
-        auto tranHashes = mbptr->GetTranHashes();
-        for (const auto& tranHash : tranHashes) {
-          TxBodySharedPtr tx;
-          if (!BlockStorage::GetBlockStorage().GetTxBody(tranHash, tx)) {
-            LOG_GENERAL(WARNING, "Missing Tx: " << tranHash);
-            result = false;
-            if (!continueOnError) {
-              break;
+        if (LOOKUP_NODE_MODE) {
```

Comment:
So when we validate lookup DB we need to enable the `LOOKUP_NODE_MODE` option?

Reply:
when the node is full node (i.e. lookup l2lookup)

Reply:
Since validatePersistence.sh also fetches the constants.xml for the specific node, this should be OK (i.e., no need to manually enable the option)

Reply:
@KaustubhShamshery this would mean a non-lookup still at least attempts to fetch all the microblocks.  That would make the resulting log file unreadable with all the "Missing MB" messages.  Perhaps we can skip looping through `txBlocks` entirely for non-lookup.